src/lib/installation/clients/desktop_finish.rb

```
@@ -31,230 +31,97 @@
 
-                @default_dm = Ops.get_string(@dd_map, [d, "logon"], "")
-                Builtins.y2milestone(
-                  "Setting logon manager %1 - package selected",
-                  @default_dm
-                )
+    def title
+      _("Initializing default window manager...")
+    end
 
-                @default_wm = Ops.get_string(@dd_map, [d, "desktop"], "")
-                Builtins.y2milestone(
-                  "Setting window manager %1 - package selected",
-                  @default_wm
-                )
+    def modes
+      [:installation, :autoinst]
+    end
 
-                @default_cursor = Ops.get_string(
-                  @dd_map,
-                  [d, "cursor"],
-                  @default_cursor
-                )
-                Builtins.y2milestone(
-                  "Setting cursor theme %1 - package selected",
-                  @default_cursor
-                )
-              else
-                Builtins.y2milestone(
-                  "Package %1 for desktop %2 neither selected nor installed, trying next desktop...",
-                  package,
-                  d
-                )
-              end
-            end
-          end
+    def write
+      selected_desktop = DefaultDesktop.Desktop
+      log.info "Selected desktop: #{selected_desktop}"
 
-          Builtins.y2milestone("Default desktop: %1", @default_wm)
-          Builtins.y2milestone("Default logon manager: %1", @default_dm)
-          Builtins.y2milestone("Default cursor theme: %1", @default_cursor)
+      if selected_desktop.nil?
+        log.info "no desktop set, skipping."
+        return nil
+      end
 
-          SCR.Write(path(".sysconfig.windowmanager.DEFAULT_WM"), @default_wm)
-          SCR.Write(
-            path(".sysconfig.windowmanager.X_MOUSE_CURSOR"),
-            @default_cursor
-          )
-          SCR.Write(path(".sysconfig.windowmanager"), nil)
+      desktop_map = DefaultDesktop.GetAllDesktopsMap[selected_desktop]
```

Maybe raise an exception if there is no desktop_map for a selected_desktop?